lmcache/v1/lookup_client/chunk_statistics_lookup_client.py

```diff
@@ -0,0 +1,175 @@
+        with self.lock:
+            self.enabled = False
+
+    def reset_statistics(self) -> None:
+        self.record_strategy.wait_for_async_processing(timeout=5.0)
+        with self.lock:
+            self.request_seen.clear()
+            self.record_strategy.reset()
+
+    def get_statistics(self) -> dict:
+        self.record_strategy.wait_for_async_processing(timeout=5.0)
+        with self.lock:
+            strategy_stats = self.record_strategy.get_statistics()
+            total_time = self.lookup_time + self.record_time + self.check_exit_time
+            overhead_time = self.record_time + self.check_exit_time
+            overhead_pct = (
+                (overhead_time / total_time * 100.0) if total_time > 0 else 0.0
+            )
+            result = {
+                "enabled": self.enabled,
+                "total_requests": len(self.request_seen),
+                "timing": {
+                    "lookup_time_seconds": self.lookup_time,
+                    "record_statistics_time_seconds": self.record_time,
+                    "check_exit_conditions_time_seconds": self.check_exit_time,
+                    "total_time_seconds": total_time,
+                    "overhead_time_seconds": overhead_time,
+                    "overhead_percentage": overhead_pct,
+                },
+                "total_chunks": strategy_stats.get("total_chunks", 0),
+                "unique_chunks": strategy_stats.get("unique_chunks", 0),
+                "duplicate_chunks": strategy_stats.get("duplicate_chunks", 0),
+                "reuse_rate": strategy_stats.get("reuse_rate", 0.0),
+                **{
+                    k: v
+                    for k, v in strategy_stats.items()
+                    if k in ("bloom_filter", "async_queue", "file_hash")
+                },
+            }
+            return result
+
+    def wait_for_async_processing(self, timeout: float = 5.0) -> bool:
+        return self.record_strategy.wait_for_async_processing(timeout)
+
+    def lookup(
+        self,
+        token_ids: Union[torch.Tensor, list[int]],
+        lookup_id: str,
+        request_configs: Optional[dict] = None,
+    ) -> Optional[int]:
+        start_time = time.time()
+        result = self.actual_lookup_client.lookup(token_ids, lookup_id, request_configs)
+        with self.lock:
+            self.lookup_time += time.time() - start_time
+        if self.enabled:
+            start_time = time.time()
+            with self.lock:
+                if lookup_id in self.request_seen:
+                    return result
+                self.request_seen.add(lookup_id)
+            self.record_strategy.record(token_ids, lookup_id)
+            with self.lock:
+                self.record_time += time.time() - start_time
+            start_time = time.time()
+            self._check_exit_conditions()
+            with self.lock:
+                self.check_exit_time += time.time() - start_time
+        return result
```

Comment:
nit: reordering the code a bit for better readability

```suggestion
        if not self.enabled or lookup_id:
            return result
        with self.lock:
            if lookup_id in self.request_seen:
                return result
        
        start_time = time.time()
        self.request_seen.add(lookup_id)
        self.record_strategy.record(token_ids, lookup_id)
        with self.lock:
            self.record_time += time.time() - start_time
            
        start_time = time.time()
        self._check_exit_conditions()
        with self.lock:
            self.check_exit_time += time.time() - start_time
        return result
```

Reply:
Done, referent this but did a minor change.